### PR TITLE
Product description AI: show a badge in "add product" store onboarding task behind a feature flag

### PR DESCRIPTION
**Pull request by jaclync**
Part of #9465 

## Why

For i2 of product description AI, we're making the feature more discoverable for new merchants first, then pre-existing merchants. In the "Add your first product" onboarding task, we're showing a badge about the AI-generated content feature.

## How

Since I'm not sure if we can ship this part of i2 this week, I created a feature flag `productDescriptionAIFromStoreOnboarding` for the emphasis on new stores.

- In `StoreOnboardingTaskViewModel`, an optional property `badgeText: String?` was added. A badge text is shown for the "Add product" onboarding text when the store is eligible for description AI. Then in `StoreOnboardingTaskView`, a badge view is shown when a task's `badgeText` is non-nil
- In `StoreOnboardingViewModel`, `isEligibleForProductDescriptionAI` to `StoreOnboardingTaskViewModel` is determined by the new feature flag, and `ProductFormAIEligibilityChecker`
- `BadgeView` component is currently shown in fixed colors. In order to customize the text & background colors, a `Customizations` struct was added with default values

### WPCOM store

Please note that the dark mode color of the badge is currently blended into the background. I asked a question in design: NMncL2ViLjomfLhuyEhVM1-fi-229:69465

- Log in to a WPCOM store that still has some pending onboarding tasks --> on the My store tab, the store setup card should be shown. if the "Add your first product" task is already complete, tap `View all` to see it in the expanded state. The "Add your first product" task should have a badge about the AI content generator

### non-WPCOM store

- Log in to a non-WPCOM store that still has some pending onboarding tasks --> on the My store tab, the store setup card should be shown. if the "Add your first product" task is already complete, tap `View all` to see it in the expanded state. The "Add your first product" task should **not** have a badge about the AI content generator

## Screenshots

non-WPCOM store | WPCOM store - light | WPCOM store - dark
-- | -- | --
Simulator Screen Shot - iPhone 14 - 2023-05-08 at 10 08 01 | Simulator Screen Shot - iPhone 14 - 2023-05-08 at 21 29 50 | Simulator Screen Shot - iPhone 14 - 2023-05-08 at 21 29 34

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/Classes/ViewRelated/Dashboard/Onboarding/StoreOnboardingTaskViewModel.swift [hunk 1]
```diff
@@ -7,45 +7,53 @@ struct StoreOnboardingTaskViewModel: Identifiable, Equatable {
...
     let title: String
     let subtitle: String
+    let badgeText: String?
 
     var isComplete: Bool {
         task.isComplete
     }
 
-    init(task: StoreOnboardingTask) {
+    init(task: StoreOnboardingTask, isEligibleForProductDescriptionAI: Bool) {
```
Comment (itsmeichigo) [inline WooCommerce/Classes/ViewRelated/Dashboard/Onboarding/StoreOnboardingTaskViewModel.swift]: Nit: `isEligibleForProductDescriptionAI` seems to be irrelevant to task types other than `.addFirstProduct`. Should we make this an associated value of that type, or move the check to this class instead?

**Review by itsmeichigo — APPROVED** [review]
LGTM 👍

File: WooCommerce/Classes/ViewRelated/Dashboard/Onboarding/StoreOnboardingTaskViewModel.swift (hunk 1 above)
Comment (jaclync) [reply WooCommerce/Classes/ViewRelated/Dashboard/Onboarding/StoreOnboardingTaskViewModel.swift]: > Should we make this an associated value of that type

The task type `TaskType` is directly decoded from the network response from `init(from:)` with a decoder, it doesn't seem straightforward to pass the condition there.

> or move the check to this class instead

If I understood this correctly, did you mean instantiating `ProductFormAIEligibilityChecker` within the view model? If so, then we need to pass `featureFlagService` and `Site` to the view model, which exposes the implementation detail of the eligibility checker.

Because the store onboarding naturally doesn't have a direct relationship with product form AI, this knowledge of product description AI eligibility has to be integrated with store onboarding somehow. Lemme know if you think I misunderstood something / any other suggested approaches.

Comment (itsmeichigo) [reply WooCommerce/Classes/ViewRelated/Dashboard/Onboarding/StoreOnboardingTaskViewModel.swift]: Thanks for the explanation! I understand it's not straightforward to inject `isEligibleForProductDescriptionAI` in another way. Another idea I can think of is to inject `badgeText` directly into the task view model with the default `nil` value - does that sound better?

Comment (jaclync) [reply WooCommerce/Classes/ViewRelated/Dashboard/Onboarding/StoreOnboardingTaskViewModel.swift]: Sure, I made some updates in https://github.com/woocommerce/woocommerce-ios/pull/9659/commits/b13a6c6e9468e683273426b4edc16b5f1b32b5d3 - I kept the localization string constant in the task view model just to have all strings in one place.

**Comment by jaclync** [comment]
Updated the PR to respond to the suggestion, plus an update on the dark mode color for the badge background in https://github.com/woocommerce/woocommerce-ios/pull/9659/commits/a99100a7e91f2b82187e10b3cc092211d264f2b2 after design confirmation NMncL2ViLjomfLhuyEhVM1-fi-229:69465 (updated the screenshots as well). Ready for another look!

**Review by itsmeichigo — APPROVED** [review]
Thanks for the update Jaclyn, everything looks good now 👍
